File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Symbols/SynthesizedMyGroupCollectionPropertyGetAccessorSymbol.cs
namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
{
	internal class SynthesizedMyGroupCollectionPropertyGetAccessorSymbol : SynthesizedMyGroupCollectionPropertyAccessorSymbol
	{
		public override bool IsSub => false;

		public override MethodKind MethodKind => MethodKind.PropertyGet;

		public override TypeSymbol ReturnType => base.PropertyOrEvent.Type;

		public SynthesizedMyGroupCollectionPropertyGetAccessorSymbol(SourceNamedTypeSymbol container, SynthesizedMyGroupCollectionPropertySymbol property, string createMethod)
			: base(container, property, createMethod)
		{
		}

		protected override string GetMethodBlock(string fieldName, string createMethodName, string targetTypeName)
		{
			return "Get\r\n" + fieldName + " = " + createMethodName + "(Of " + targetTypeName + ")(" + fieldName + ")\r\nReturn " + fieldName + "\r\nEnd Get\r\n";
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Symbols/SynthesizedMyGroupCollectionPropertyGetAccessorSymbol.vb
Imports Microsoft.CodeAnalysis
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
	Friend Class SynthesizedMyGroupCollectionPropertyGetAccessorSymbol
		Inherits SynthesizedMyGroupCollectionPropertyAccessorSymbol
		Public Overrides ReadOnly Property IsSub As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property MethodKind As Microsoft.CodeAnalysis.MethodKind
			Get
				Return Microsoft.CodeAnalysis.MethodKind.PropertyGet
			End Get
		End Property

		Public Overrides ReadOnly Property ReturnType As TypeSymbol
			Get
				Return MyBase.PropertyOrEvent.Type
			End Get
		End Property

		Public Sub New(ByVal container As SourceNamedTypeSymbol, ByVal [property] As SynthesizedMyGroupCollectionPropertySymbol, ByVal createMethod As String)
			MyBase.New(container, [property], createMethod)
		End Sub

		Protected Overrides Function GetMethodBlock(ByVal fieldName As String, ByVal createMethodName As String, ByVal targetTypeName As String) As String
			Return [String].Concat(New [String]() { "Get" & VbCrLf & "", fieldName, " = ", createMethodName, "(Of ", targetTypeName, ")(", fieldName, ")" & VbCrLf & "Return ", fieldName, "" & VbCrLf & "End Get" & VbCrLf & "" })
		End Function
	End Class
End Namespace